csp/sync.csp

```diff
@@ -0,0 +1,184 @@
+                    </div>
+                    <div style="display: #(noFileDisplay)#">
+                        <h2 class="text-center">No files to commit with sync</h2>
+                    </div>
+                    <h3 class="section-header">Sync details:</h3>
+                    <p>Upon syncing, the local repository will pull the changes from remote and commit the newest changes before committing and pushing</p>
+                    <server>
+                        
+                        if settings.defaultMergeBranch '= "" {
+                            &html<<p>Local changes will be merged with #(..EscapeHTML(settings.defaultMergeBranch))# after receiving the latest changes</p>>
+                        }
+                    </server>
+
+                    <button class="btn btn-lg btn-primary" id="syncBtn" onClick="disableInput(); #server(..PerformSync(self.document.getElementById('syncMsg').value))#">Sync</button>
+                    <div>
+                        <h3 class="output-header" id="outputHeader" style="display: #(outputDisplay)#">Sync output: </h3>
+                        <div class="container output" id="outputContainer" style="display: #(outputDisplay)#">
+                            <pre id="outputBox"></pre>
+                        </div>
+                    </div>
+                </div>
+            </div>
+            
+        </div>
+    </body>
+    <script type="text/javascript">
+    function disableInput() {
+        document.getElementById('syncMsg').disabled = true;
+        document.getElementById('syncBtn').innerHTML = 'Syncing...';
+        document.getElementById('syncBtn').disabled = true;
+    }
+    </script>
+    <script language="cache" method="PerformSync" arguments="syncMsg:%String">
+        &js<document.getElementById('outputContainer').style.display = 'block'>
+        &js<document.getElementById('outputHeader').style.display = 'block'>
+        set buffer = ##class(SourceControl.Git.Util.Buffer).%New()
+        do buffer.BeginCaptureOutput()
+
+        set st = ##class(SourceControl.Git.Utils).Sync(syncMsg)
+
+        set out = ##class(%Stream.GlobalCharacter).%New()
+        do buffer.EndCaptureOutput(.out)
+        while 'out.AtEnd {
+            set line = out.ReadLine()
+            set escapedLine = ..EscapeHTML(line)
+            set escapedLine = $replace(escapedLine, $char(10), "<br>")
+
+            &js<
+                var outputContainer = document.getElementById('outputBox');
+                var lineText = "#(escapedLine)#";
+                var lineTextNode = document.createTextNode(lineText);
+                outputContainer.innerHTML += lineText + "<br>";
+            >
+        }
+        &js<document.getElementById('syncBtn').innerHTML = 'Synced'>
+    </script>
+</html>
+<script method='OnPreHTTP' language='cache' runat='server' returntype='%Boolean'>	
+	try {
+		set %session.UseSessionCookie = 1 // Always set back to autodetect
```

Comment:
It would be worth refactoring this to a shared method somewhere since many places use it. Doesn't need to happen in this PR though.